@xen-orchestra/fs/src/fs.spec.js

```diff
@@ -310,5 +310,17 @@ handlers.forEach(url => {
         await handler.unlink('file')
       })
     })
+
+    describe('#write()', () => {
+      beforeEach(() => handler.outputFile('file', TEST_DATA))
+
+      testWithFileDescriptor('file', 'r+', async ({ file }) => {
+        const WRITE_SIZE = 256
+        const WRITE_TEST_DATA = unsecureRandomBytes(WRITE_SIZE)
+        await handler.write(file, WRITE_TEST_DATA, 0)
```

Comment:
Working at a non-0 offset would be a better test.

You can even use a random offset.